src/MCPClient/lib/clientScripts/upload-archivesspace.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/python
+
+import argparse
+import logging
+import os
+from urlparse import urljoin
+
+from main.models import ArchivesSpaceDIPObjectResourcePairing
+
+# archivematicaCommon
+from archivesspace.client import ArchivesSpaceClient
+from elasticSearchFunctions import getDashboardUUID
+from xml2obj import mets_file
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+logger.addHandler(logging.FileHandler('/tmp/as_upload.log', mode='a'))
+
+
+def recursive_file_gen(mydir):
+    for root, dirs, files in os.walk(mydir):
+        for file in files:
+            yield os.path.join(root, file)
+
+
+def get_files_from_dip(dip_location, dip_name, dip_uuid):
+    # need to find files in objects dir of dip:
+    # go to dipLocation/dipName/objects
+    # get a directory listing
+    # for each item, set fileName and go
+    try:
+        mydir = dip_location + "objects/"
+        mylist = list(recursive_file_gen(mydir))
+
+        if len(mylist) > 0:
+            return mylist
+        else:
+            logger.error("no files in " + mydir)
+            raise ValueError("cannot find dip")
+            exit(2)
```

Comment:
Will exit ever get called? Won't the raise trigger first and end the function?  Same below.